tests/tests/unit-tests/routers/UserRouterTest.js

```diff
@@ -43,18 +45,65 @@ describe ('UserRouter', function () {
               return done (err);
             }
 
-            access_token = res.body.token;
+            adminAccessToken = res.body.token;
             return done ();
           });
       });
     });
 
-    describe ('POST', function (done) {
+    describe ('Authentication', function () {
+
+      var userAccessToken;
+
+      before (function (done) {
+        userData = users[0];
+        var User = blueprint.app.models.User;
+        var newUser = new User (userData);
+
+        newUser.save (function (err, user) {
+          if (err) { return done (err); }
+
+          var data = {
+            username: user.username,
+            password: user.password
+          }
+
+          request (blueprint.app.server.app)
+            .post ('/login')
+            .send (data)
+            .expect (200)
+            .end (function (err, res) {
+              if (err) {
+                return done (err);
+              }
+
+              userAccessToken = res.body.token;
+              return done ();
+            });
+        });
+      });
+
+      it ('should allow admin to access user routes', function (done) {
+        request (blueprint.app.server.app)
+          .get ('/v1/admin/users') // route
+          .set ('Authorization', 'bearer ' + adminAccessToken)
+          .expect (200, done);
+      });
+
+      it ('should not allow user to access user routes', function (done) {
+        request (blueprint.app.server.app)
+          .get ('/v1/admin/users') // route
+          .set ('Authorization', 'bearer ' + userAccessToken)
+          .expect (401, done);
+      });
+    });
+
+    describe ('POST', function () {
       it ('should create a user in the database', function (done) {
         request (blueprint.app.server.app)
-          .post ('/v1/users') // route
-          .set ('Authorization', 'bearer ' + access_token)
-          .send (userData) // data being sent
+          .post ('/v1/admin/users') // route
```

Comment:
It is defined in the /v1/IndexRouter.js file.

If you look at the Router directory, you will see a sub directory called admin within the v1 directory.  What this allows us to do is have nested authentication logic.

Inside the /v1/IndexRouter.js file I have declared that anyone trying to access those routes must also pass the isAdminToken middleware function.  This will take the incoming authenticated token and see if the user it is associated with has the admin role.  If it does, than it can access the routes inside the admin folder (/users).  Else it cannot.  This way we can put routes that all users can access in just /v1 but admin specific routes in /v1/admin/.